File: Stacks/stack_array.c
#include<stdio.h>
#include<stdlib.h>
#include<limits.h>

struct Stack{
	int top;
	unsigned capacity;
	int* array;
};

struct Stack* createstack(unsigned capacity){
	struct Stack* stack = (struct Stack*)malloc(sizeof(struct Stack));
	stack-> capacity = capacity;
	stack-> top = -1;
	stack->array = (int*)malloc(stack->capacity * sizeof(int));
	return stack;
}

int isFull(struct Stack* stack){
	return stack->top == stack->capacity-1;
}
int isEmpty(struct Stack* stack){
	return stack->top == -1;
}
void push(struct Stack* stack, int value){
	if(isFull(stack))
		return;
	stack->array[++stack->top] = value;
	printf("%d pushed into stack\n",value);
}
int pop(struct Stack* stack){
	if(isEmpty(stack))
		return INT_MIN;
	return stack->array[stack->top--];
}

int main(){
	struct Stack* stack = createstack(100);

	push(stack, 1);
	push(stack, 2);
	push(stack, 3);

	printf("%d popped from the stack \n", pop(stack));
	printf("%d popped from the stack \n", pop(stack));
	printf("%d popped from the stack \n", pop(stack));
}

File: stack_array.py
from sys import maxsize

def createstack():
	stack = []
	return stack

def isEmpty(stack):
	return len(stack) == 0

def push(stack , value):
	stack.append(value)
	print(str(value) + " pushed into stack")

def pop(stack):
	if isEmpty(stack):
		return 	str(-maxsize-1)

	return str(stack.pop())

stack = createstack()

push(stack, 1)
push(stack, 2)
push(stack, 3)

print( pop(stack) + " popped from stack")
print( pop(stack) + " popped from stack")
print( pop(stack) + " popped from stack")